zerver/models.py

```diff
@@ -3562,8 +3562,10 @@ class Meta(AbstractUserMessage.Meta):
                 "user_profile",
                 "message",
                 condition=Q(flags__andnz=AbstractUserMessage.flags.mentioned.mask)
-                | Q(flags__andnz=AbstractUserMessage.flags.wildcard_mentioned.mask),
-                name="zerver_usermessage_wildcard_mentioned_message_id",
+                | Q(flags__andnz=AbstractUserMessage.flags.stream_wildcard_mentioned.mask)
+                | Q(flags__andnz=AbstractUserMessage.flags.topic_wildcard_mentioned.mask)
+                | Q(flags__andnz=AbstractUserMessage.flags.group_mentioned.mask),
+                name="zerver_usermessage_any_mentioned_message_id",
```

Comment:
Let's use this as the main thread for discussing how we want this index to work... or maybe a chat.zulip.org thread pointing at this. The question in my mind is whether it would be a better index to instead index on:
`condition=Q(flags__andnz=AbstractUserMessage.flags.mentioned.mask | AbstractUserMessage.flags.stream_wildcard_mentioned.mask | AbstractUserMessage.flags.topic_wildcard_mentioned.mask | AbstractUserMessage.flags.group_mentioned.mask)`, so that we're asking the database to do a single bitmask check, rather than 4 parallel ones. 

See https://github.com/zulip/zulip/pull/26382/commits/4f550840f35ac9ee3dad9d404d2056f90f521507#r1349285843 for evidence of the precise index/query that this implementation generates.